docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,440 @@
+
+```python
+creative_space = TextSimilaritySpace(
+    text=user.ad_creative, model="sentence-transformers/all-mpnet-base-v2"
+)
+
+activity_space = NumberSpace(
+    number=user.activity, mode=Mode.SIMILAR, min_value=0.0, max_value=1.0
+)
+
+recency_space = RecencySpace(
+    timestamp=user.signup_date,
+    period_time_list=[PeriodTime(timedelta(days=65)), PeriodTime(timedelta(days=185))],
+    negative_filter=0.0,
+)
+```
+
+Now, let's plot our recency scores by date.
+
+```python
+recency_plotter = RecencyPlotter(recency_space, context_data=EXECUTOR_DATA)
+recency_plotter.plot_recency_curve()
+```
+
+![](../assets/use_cases/user_acquisition_analytics/recency_scores-by-date.png)
+
+[how exactly do the recency scores work?] 
+
+Next, we set up an in-memory data processing pipeline for indexing, parsing, and executing operations on user data, including clustering (where RecencySpace lets our model take account of user signup recency).
+
+First, we create our index with the spaces we use for clustering.
+
+```python
+user_index = Index(spaces=[creative_space, activity_space, recency_space])
+```
+
+Now for dataframe parsing.
+
+```python
+user_df_parser = DataFrameParser(schema=user)
+```
+
+We create an `InMemorySource` object to hold the user data in memory, and set up our executor (with our user data source and index) so that it takes account of context data.
+
+```python
+source_user: InMemorySource = InMemorySource(user, parser=user_df_parser)
+executor: InMemoryExecutor = InMemoryExecutor(
+    sources=[source_user], indices=[user_index], context_data=EXECUTOR_DATA
+)
+app: InMemoryApp = executor.run()
+```
+
+And then input our user data. (This step make take a few minutes. In the meantime, why not read more about vectors in [Vectorhub](https://superlinked.com/vectorhub/).)
+
+```python
+source_user.put([user_df])
+```
+
+## Load features
+
+Next, we collect all our vectors from the app.
+
+```python
+vs = VectorSampler(app=app)
+vector_collection = vs.get_all_vectors(user_index, user)
+vectors = vector_collection.vectors
+vector_df = pd.DataFrame(vectors, index=[int(id_) for id_ in vector_collection.id_list])
+vector_df.head()
+```
+
+```python
+vector_df.shape
+```
+
+Here are the first five rows (of 8000), and 776 columns, of the resulting dataframe:
+
+![](../assets/use_cases/user_acquisition_analytics/collected_vectors.png)
+
+## Clustering
+
+Next, we fit a clustering model.
+
+NOTE: If you run into issues running this notebook on Colab, we suggest using your own environment. In Colab, the management of python packages is less straight-forward, which can cause issues.
+
+```python
+hdbscan = HDBSCAN(min_cluster_size=500, metric="cosine")
+hdbscan.fit(vector_df.values)
+```
+
+workaround on colab
```

Comment:
@morkapronczay
This makes the umap look different but more or less approximates the blobs of the original. Should we include this for anyone who wants to run in colab?

Reply:
I suppose we can leave this in the article, but would not put it into the notebook yet. We should discuss how much we want to support Colab, I'll bring this up internally.

Reply:
sounds good!

Reply:
the Colab issue for now is solved (they updated the sklearn version) and we are also preparing a longer term safeguard too. All is going well :)